javascript/hangman.js

```diff
@@ -1,35 +1,65 @@
 class Hangman {
   constructor(words) {
     this.words = words;
-    // ... your code goes here
+    this.secretWord = this.pickWord();
+    this.letters = [];
+    this.guessedLetters = '';
+    this.errorsLeft = 10;
   }
 
   pickWord() {
-    // ... your code goes here
+    return this.secretWord = this.words[Math.floor(Math.random() * this.words.length)];
   }
 
   checkIfLetter(keyCode) {
-    // ... your code goes here
+    if (keyCode >= 65 && keyCode <= 90) {
+      return true;
+    } else {
+      return false;
+    }
   }
 
   checkClickedLetters(letter) {
-    // ... your code goes here
+    if (this.letters.includes(letter)) {
+      return false;
+    } else {
+      return true;
+    }
   }
 
   addCorrectLetter(letter) {
-    // ... your code goes here
-  }
-
+    this.guessedLetters += letter;
+  } 
+  
   addWrongLetter(letter) {
-    // ... your code goes here
+    this.errorsLeft--;
+
+    if (!this.letters.includes(letter)){
+      this.letters.push(letter);
+
+    }
+
   }
 
   checkGameOver() {
-    // ... your code goes here
+    if( this.errorsLeft > 0) {
+      return false;
+    }
+    else {
+      return true;
+    }
+    
```

Comment:
Podriamos retornar `return !this.errorsLeft`? Para refactorizar, puedes probarlo y ver como se haria tambien en el resto de las funciones, seria aplicar un poco de lo que hemos visto en el lab de koans.